src/it/scala/net/snowflake/spark/snowflake/SnowflakeResultSetRDDSuite.scala

```diff
@@ -0,0 +1,402 @@
+                  | '2014-01-11 06:12:13.123', '2014-01-11 06:12:13.123456'
+                  | ),(
+                  | 2,
+                  | '2014-01-11 06:12:13.123456', '2014-01-11 06:12:13',
+                  | '2014-01-11 06:12:13.123', '2014-01-11 06:12:13.123456',
+                  |
+                  | '2014-01-11 06:12:13.123456', '2014-01-11 06:12:13',
+                  | '2014-01-11 06:12:13.123', '2014-01-11 06:12:13.123456',
+                  |
+                  | '2014-01-11 06:12:13.123456', '2014-01-11 06:12:13',
+                  | '2014-01-11 06:12:13.123', '2014-01-11 06:12:13.123456'
+                  | )""".stripMargin)
+
+    val tmpdf = sparkSession.read
+      .format(SNOWFLAKE_SOURCE_NAME)
+      .options(thisConnectorOptionsNoTable)
+      .option("dbtable", s"$test_table_timestamp")
+      .load()
+
+    tmpdf.createOrReplaceTempView("test_table_timestamp")
+  }
+
+  private val largeStringValue =
+    s"""spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |spark_connecor_test_large_result_1234567890
+       |""".stripMargin.filter(_ >= ' ')
+
+  private def setupLargeResultTable() : Unit = {
+    jdbcUpdate(s"""create or replace table $test_table_large_result (
+                  | int_c int, c_string string(1024) )""".stripMargin)
+
+    jdbcUpdate(s"""insert into $test_table_large_result select
+                  | seq4(), '$largeStringValue'
+                  | from table(generator(rowcount => 100000))""".stripMargin)
+
+    val tmpdf = sparkSession.read
+      .format(SNOWFLAKE_SOURCE_NAME)
+      .options(thisConnectorOptionsNoTable)
+      .option("dbtable", s"$test_table_large_result")
+      .load()
+
+    tmpdf.createOrReplaceTempView("test_table_large_result")
+  }
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    // There is bug for Date.equals() to compare Date with different timezone,
+    // so set up the timezone to work around it.
+    val gmtTimezone = TimeZone.getTimeZone("GMT")
+    TimeZone.setDefault(gmtTimezone)
+
+    connectorOptionsNoTable.foreach(tup => { thisConnectorOptionsNoTable += tup })
+
+    // Setup special options for this test
+    thisConnectorOptionsNoTable += ("use_copy_unload" -> "false")
+    thisConnectorOptionsNoTable += ("partition_size_in_mb" -> "50")
+    thisConnectorOptionsNoTable += ("query_result_format" -> "arrow")
+    thisConnectorOptionsNoTable += ("time_output_format" -> "HH24:MI:SS.FF")
+
+    setupNumberTable()
+    setupStringBinaryTable()
+    setupDateTimeTable()
+    setupTimestampTable()
+    setupLargeResultTable()
+
+    SnowflakeConnectorUtils.enablePushdownSession(sparkSession)
```

Comment:
I think we don't need to turn on pushdown for this test suite, and we don't really care about the query generated.

Reply:
All the query is SELECT * FROM <TABLE>. It will be deleted.